reactor-netty-http/src/main/java/reactor/netty/http/server/HttpServer.java

```diff
@@ -713,9 +741,34 @@ public final HttpServer secure(Consumer<? super SslProvider.SslContextSpec> sslP
 	 * @return a new {@link HttpServer}
 	 */
 	public final HttpServer secure(SslProvider sslProvider) {
+		return secure(sslProvider, false);
+	}
+
+	/**
+	 * Applies an SSL configuration via the passed {@link SslProvider}.
+	 * <p>
+	 * If {@link SelfSignedCertificate} needs to be used, the sample below can be
+	 * used. Note that {@link SelfSignedCertificate} should not be used in production.
+	 * <pre>
+	 * {@code
+	 *     SelfSignedCertificate cert = new SelfSignedCertificate();
+	 *     SslContextBuilder sslContextBuilder =
+	 *             SslContextBuilder.forServer(cert.certificate(), cert.privateKey());
+	 *     secure(sslContextSpec -> sslContextSpec.sslContext(sslContextBuilder), true);
+	 * }
+	 * </pre>
+	 *
+	 * @param sslProvider         The provider to set when configuring SSL
+	 * @param redirectHttpToHttps true enables redirecting HTTP to HTTPS on the same port.
+	 *                            This configuration is applicable only for HTTP/1.x.
+	 * @return a new {@link HttpServer}
+	 * @since 1.0.5
+	 */
+	public final HttpServer secure(SslProvider sslProvider, boolean redirectHttpToHttps) {
```

Comment:
What about instead of the boolean args in two places, having a single method like in `HttpServerConfig`, something like `secureRedirectHttpToHttps(boolean)`? Boolean args (and overloading) can become challenging in the future if more need to be added. I'm thinking such a method could be expected to be called after `secure` and be rejected if secure wasn't called already. Just a thought.

Reply:
@rstoyanchev Till now we do not have a requirement for ordering of the configuration, because of that we handle this when the respective configuration is made.
For example when we configure the HTTP/1.1 and HTTP/2 protocols and the default SslProvider when on the client side, we need to choose the correct default SslProvider depending on the protocol:

https://github.com/reactor/reactor-netty/blob/f8c75b9fea8d378cf93f7696ead8a2a40c750f2f/reactor-netty-http/src/main/java/reactor/netty/http/client/HttpClient.java#L1268

https://github.com/reactor/reactor-netty/blob/f8c75b9fea8d378cf93f7696ead8a2a40c750f2f/reactor-netty-http/src/main/java/reactor/netty/http/client/HttpClient.java#L1340

So the code will become a little bit more complicated.

Reply:
@rstoyanchev Both using an independent method `redirectHttpToHttps` or using `redirectHttpToHttps` as a param in `secure` are reasonable. And I have discussed this with @violetagg. Personally, I am okay with both for the following reasons:

* using an independent method `redirectHttpToHttps` for a more fluent API, especially `secure(SslProvider sslProvider, boolean redirectHttp2Https=false)`, is usually used like `secure(sslContextSpec -> sslContextSpec.sslContext(sslContextBuilder), true)` and it's a little long and isn't that fluent compared to add redirectHttp2Https as the second parameter of secure method.

* It's also reasonable for redirectHttp2Https to be a parameter of secure(SslProvider sslProvider, boolean redirectHttp2Https=false) because redirectHttp2Https can only work when ssl is enabled.

@violetagg wdyt?

Reply:
@rstoyanchev As long term solution I prefer to think about `SslProvider` configurations per different protocol.
@JamesChenX Do you need this feature for both methods `secure(SslProvider sslProvider, true)` and `secure(sslContextSpec -> sslContextSpec.sslContext(sslContextBuilder), true)` or we can limit the configuration to one of them? @rstoyanchev wdyt?

Reply:
Feel free to decide how to best proceed. Overloading only one of the methods is an option as well.

Reply:
@JamesChenX shall we go with a reduced scope with only one method or to leave it as it is?

Reply:
@violetagg I prefer `secure(sslContextSpec -> sslContextSpec.sslContext(sslContextBuilder), true)`, and it's just a choice between "Fluent API" (one for `secure` and another for `redirectHttpToHttps`) and "Code clarity" (uses one method for the these two).

The current style `secure(sslContextSpec -> sslContextSpec.sslContext(sslContextBuilder), true)` is less confusing because the method indicates that `redirectHttpToHttps` can only work when SSL is enabled.

Reply:
@JamesChenX Agree
Ok let's reduce the scope and expose only `secure(sslContextSpec -> sslContextSpec.sslContext(sslContextBuilder), true)`
Can you add this change to the PR and I think after that we are ready to merge the change.

Reply:
@violetagg I misinterpreted your words, and we can just keep the current commit with both the existing method `secure(SslProvider sslProvider)` and the new method `secure(SslProvider sslProvider, boolean redirectHttpToHttps)` without making an unnecessary breaking change.